features/nvstore/source/nvstore.h

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NVSTORE_H
+#define __NVSTORE_H
+
+// These addresses need to be configured according to board (in mbed_lib.json)
+#ifndef DEVICE_FLASH
+#undef NVSTORE_ENABLED
+#define NVSTORE_ENABLED 0
+#endif
+
+#if NVSTORE_ENABLED
+#include <stdint.h>
+#include <stdio.h>
+#include "platform/NonCopyable.h"
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+
+enum {
```

Comment:
Can this enum be named?

Reply:
Can do. Although I felt that having APIs returning integers instead of this enum would be more convenient for the user. Saw this practice used in few of the block devices.